web_environment_ribbon/static/src/js/ribbon_show.js

```diff
@@ -0,0 +1,16 @@
+/* Copyright 2015 Sylvain Calador <sylvain.calador@akretion.com>
+   Copyright 2015 Javi Melendez <javi.melendez@algios.com>
+   Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+   Copyright 2017 Thomas Binsfeld <thomas.binsfeld@acsone.eu>
+   Copyright 2017 Xavier Jiménez <xavier.jimenez@qubiq.es>
+   Copyright 2018 David Arnold <dar@xoe.solutions>
+   License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+odoo.define('web_environment_ribbon.ribbon_show', function(require) {
+"use strict";
+
+    var core = require('web.core');
+    var showRibbon = require('web_environment_ribbon.ribbon');
+    core.bus.on('web_client_ready', null, showRibbon);
```

Comment:
I don't think this operation justifies the creation of a whole new module.

This patch could be way smaller by just putting this line at the end of the original JS module, and adding `return {showRibbon: showRibbon}` to the end. Could you do that please?

Reply:
I'm not sure if I understand how that would fit the purpose of the whole exercise.
Could you explain?
What is the goal? -> split our _rendering_ so that I can customize the rendering, not the renderer.
Or could you give an example how to achieve that more cleanly?

My line of thought: I can override the loading with an xpath and xml inheritance and therefore overload this extra module with my custom _rendering_ function, still calling the same _renderer_.

Please let me know, if this use case is intelligible? - :wink:

Reply:
Maybe I confused you with the word "module". I was referring to a JS module, not to a Odoo module.

Summarizing, you're doing this:

1. Creating a JS module `web_environment_ribbon.ribbon` that returns the method that renders the ribbon.
2. Creating a JS module `web_environment_ribbon.ribbon_show` that binds the previous method to the `web_client_ready` event.

This adds some complexity that has actually no real benefit over just:

1. Have a single JS module `web_environment_ribbon.ribbon`, as it was before.
1. Inside that module, define a named `showRibbon` function.
1. Bind that function to the `web_client_ready` event.
1. Return the function, just in case somebody wants to extend it, by adding `return {showRibbon: showRibbon}` to the end of the module.

In its simplest form, this would take a diff of about 3 LOC.

OTOH, if your ultimate goal is to override the rendering of the ribbon, I suggest you to refactor subtly the method itself and make it render a client-side Qweb template, as normal widgets do.

Reply:
Possibly even if you have the returned function, modifying it wouldn't mean modifying what was bound to the event.

I doubt you have more to do besides customizing the layout of the ribbon. Most useful would be then to use client-side qweb to render, and just override the qweb template in your addon.

Reply:
I plan to wrap a java script timer around and possibly a help text pop-up.
So if you can quickly confirm, that I can rebind on the very same event, then I can achieve the same thing the way you suggested.

I'd be happy to refactor and get this trough.